docs/user-guide/topics/readiness.md

```diff
@@ -0,0 +1,67 @@
+---
+template: main.html
+---
+
+# Checking readiness of Kubernetes resources
+
+When Iter8 experiment rely on Kubernetes resources, the readiness of those resources can be checked first.  
+
+For example, to check the readiness of a `Deployment` resource:
+
+```shell
+iter8 k -c load-test-http \
+--set url=http://httpbin.default \
+--set ready.deploy=httpbin
+```
+
+Currently, Iter8 provides native support for the following resource types:
+
+- `Deployment`: resource is ready if it exists and has the status condition `Available` set to `True`
+- `Service`: resource is ready if it exists
+
+The option `ready.timeout` can be used to specify the maximum time that the experiment should wait for the readiness condition to be satisfied.
+
+## Extending to new resource kinds
+
+1. Download chart(s):
+
+    ```shell
+    iter8 hub -c load-test-http
+    ```
+
+2. Add an entry for the new resource type in `charts/iter8lib/templates/_task.ready.tpl`.
```

Comment:
I don't think we want to describe local chart editing here ... we want people to contribute to Iter8 repo instead. 
***
1. See the current implementation for deployment and service here ... 
2. As you can see, it involves two templates (`task.readiness`, and `task.ready.rbac` .. with links of course). Edit them to add support for your resource type.
3. For example, to add support for `ksvc` (Knative service), do the following ...

Reply:
We encourage users to contribute readiness checks for various Kubernetes resources and CRDs. ... or something like that.

Reply:
This is a nice idea -- encourage pull requests.
However, it is a potentially longer path to use. Local editing is much faster to get started.

Reply:
done